### PR TITLE
Cross link RMM & libcudf Doxygen docs [skip-ci]

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR updates `cpp/doxygen/Doxyfile` to consume the generated Doxygen tags from `rmm` (see rapidsai/rmm#672). This will enable linking between the `cudf` docs and `rmm` docs.

This PR along with rapidsai/rmm#672 closes issue #5152.

I also updated `docs/cudf/source/conf.py` and added it to `update-version.sh`.

### CODE REVIEW

**Review by vuule — APPROVED** [review]
I don't fully understand what this does, approving to meet the cpp-owners requirement (no C++ changes anyway).

**Review by nvdbaranec — APPROVED** [review]
I echo Vukasin's comments :)

**Comment by ajschmidt8** [comment]
> I don't fully understand what this does, approving to meet the cpp-owners requirement (no C++ changes anyway).

> I echo Vukasin's comments :)

@vuule, @nvdbaranec this PR makes it so that when people see an `rmm` reference in the `cudf` docs (highlighted in the screenshot), they can click it and it will link them directly to the correct `rmm` documentation page.


image

File: ci/docs/build.sh [hunk 1]
```diff
@@ -40,6 +40,7 @@ conda list --show-channel-urls
 #libcudf Doxygen build
 gpuci_logger "Build libcudf docs..."
 cd $PROJECT_WORKSPACE/cpp/doxygen
+wget "https://raw.githubusercontent.com/rapidsai/docs/gh-pages/api/librmm/${BRANCH_VERSION}/rmm.tag" || echo "Failed to download rmm Doxygen tag"
```
Comment (karthikeyann) [inline ci/docs/build.sh:43]: rmm and cudf may diverge in version numbers. Check with @harrism

**Review by karthikeyann — CHANGES_REQUESTED** [review]

File: ci/docs/build.sh (hunk 1 above)
Comment (dillon-cullinan) [reply ci/docs/build.sh:43]: @karthikeyann If this is true, this should be communicated with OPS ASAP. That could affect a lot more than just doc builds.

**Review by dillon-cullinan — APPROVED** [review]

File: ci/docs/build.sh (hunk 1 above)
Comment (harrism) [reply ci/docs/build.sh:43]: There are no current plans to diverge. RMM and cuDF are released on the same schedule.

**Comment by harrism** [comment]
Fixes #5152

**Review by karthikeyann — APPROVED** [review]

**Pull request merged**